### PR TITLE
cp2k : added package

### PR DESCRIPTION
**Pull request by alalazo**
This should be a good initial version of the package. I'll add optional features (including plumed) asap.

### CODE REVIEW

**Comment by adamjstewart** [comment]
I'm working on libint (#1264) and @mdevlin1 is working on libxsmm (#1322) and libxstream (coming soon)

File: var/spack/repos/builtin/packages/cp2k/package.py [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    depends_on('blas')
+    depends_on('fftw')
+
+    depends_on('mpi', when='+mpi')
+    depends_on('scalapack', when='+mpi')
+
+    # TODO : add dependency on libint
+    # TODO : add dependency on libsmm, libxsmm
+    # TODO : add dependency on elpa
+    # TODO : add dependency on CUDA
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/cp2k/package.py]: There is actually a CUDA package already.

File: var/spack/repos/builtin/packages/cp2k/package.py [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+    # TODO : add dependency on PEXSI
+    # TODO : add dependency on QUIP
+    # TODO : add dependency on plumed
+    # TODO : add dependency on libwannier90
+
+    parallel = False
+
+    def install(self, spec, prefix):
+        # Construct a proper filename for the architecture file
+        cp2k_architecture = '{0.architecture}-{0.compiler.name}'.format(spec)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/cp2k/package.py]: I had no idea you could do this, this is so cool!

File: var/spack/repos/builtin/packages/cp2k/package.py [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+from spack import *
+
+
+class Cp2k(Package):
+    """CP2K is a quantum chemistry and solid state physics software package
+    that can perform atomistic simulations of solid state, liquid, molecular,
+    periodic, material, crystal, and biological systems
+    """
+    homepage = 'http://www.example.com'
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/cp2k/package.py]: Don't forget a homepage

File: var/spack/repos/builtin/packages/cp2k/package.py (hunk 1 above)
Comment (alalazo) [inline var/spack/repos/builtin/packages/cp2k/package.py]: I know, but I didn't go through the makefile in arch to see how they want things to be specified... I think I'll add the otional dependencies after this one will be merged.

**Comment by alalazo** [comment]
@tgamblin @becker33 @adamjstewart I think this is ready to be merged

**Pull request merged**